src/me/remie/vulcan/leaguetasks/LeagueScript.java

```diff
@@ -96,19 +98,18 @@ public void onProcess() {
     }
 
     private void checkTasks() {
-        final SimpleWidget leaguesWidget = ctx.widgets.getWidget(657, 18);
-        if (leaguesWidget == null || leaguesWidget.isHidden()) {
-            ctx.log("Opening leagues menu");
-            ctx.game.tab(Game.Tab.QUESTS);
-            ctx.menuActions.clickButton(41222167);
-            ctx.sleep(1000);
-            ctx.menuActions.clickButton(42991640);
-            ctx.onCondition(() -> {
-                final SimpleWidget leaguesWidget1 = ctx.widgets.getWidget(657, 18);
-                return leaguesWidget1 != null && !leaguesWidget1.isHidden();
-            }, 1000, 5);
+        final SimpleWidget leaguesWidget = leaguePanel.getLeaguePanelWidget();
+
+        if (!leaguePanel.isLeaguePanelOpen()) {
+            leaguePanel.openLeaguePanel();
             return;
         }
```

Comment:
```suggestion
        if (!leaguePanel.openLeaguePanel()) {
            return;
        }
```
The open function also checks if the panel is already open or not, we can just check the result of the open function isntead.